internal/fields/validate.go

```diff
@@ -692,26 +693,26 @@ func (v *Validator) validateScalarElement(key string, val any, doc common.MapStr
 		case isFlattenedSubfield(key, v.Schema):
 			return nil // flattened subfield, it will be stored as member of the flattened ancestor.
 		case isArrayOfObjects(val):
-			return fmt.Errorf(`field %q is used as array of objects, expected explicit definition with type group or nested`, key)
+			return multierror.Error{fmt.Errorf(`field %q is used as array of objects, expected explicit definition with type group or nested`, key)}
 		case couldBeMultifield(key, v.Schema):
-			return fmt.Errorf(`field %q is undefined, could be a multifield`, key)
+			return multierror.Error{fmt.Errorf(`field %q is undefined, could be a multifield`, key)}
 		case !isParentEnabled(key, v.Schema):
 			return nil // parent mapping is disabled
 		default:
-			return fmt.Errorf(`field %q is undefined`, key)
+			return multierror.Error{fmt.Errorf(`field %q is undefined`, key)}
 		}
 	}
 
 	if !v.disabledNormalization {
 		err := v.validateExpectedNormalization(*definition, val)
 		if err != nil {
-			return fmt.Errorf("field %q is not normalized as expected: %w", key, err)
+			return multierror.Error{fmt.Errorf("field %q is not normalized as expected: %w", key, err)}
 		}
 	}
 
-	err := v.parseElementValue(key, *definition, val, doc)
-	if err != nil {
-		return fmt.Errorf("parsing field value failed: %w", err)
+	errs := v.parseElementValue(key, *definition, val, doc)
+	if len(errs) > 0 {
+		return errs.Unique()
```

Comment:
One of the main changes in this PR, here it is removed the `parsing field value field` from the error (and just return all the errors that could be found).

To avoid errors in the output like:
``` 
[0] parsing field value failed: [0] field "gcp.firewall.rule_details.ip_port_info.ip_protocol" is undefined
[1] field "gcp.firewall.rule_details.ip_port_info.port_range" is undefined
[1] parsing field value failed: [0] field "gcp.firewall.rule_details.ip_port_info.port_range" is undefined
[1] field "gcp.firewall.rule_details.ip_port_info.ip_protocol" is undefined
```